Enable content-type-specific field queries in GraphQL page API

hassandotcms opened the pull request:
## Summary

  Enables GraphQL queries to access content-type-specific fields for pages through inline fragments. Fixes #32695.

## What's New

  Added a `page` field to GraphQL page queries that exposes the page as its underlying contentlet type:

  ```graphql
  query {
    page(url: "/about") {
      title
      page {
        ... on htmlpageasset {
          seokeywords
          seodescription
        }
      }
    }
  }
  ```

 ## Changes

  - Modified PageAPIGraphQLTypesProvider.java to add the new field using PropertyDataFetcher pattern
  - Added Postman tests for htmlpageasset and custom page types

 ## Why This Matters

  - Developers can now access SEO metadata and custom fields on pages
  - Works automatically with all page types (default and custom)
  - No breaking changes to existing queries

  ## Testing

  Postman tests added for:
  - SEO field access on htmlpageasset pages
  - Custom page type fields (Destination)
  - Content type resolution

hassandotcms committed:
32695-enable-support-to-query-page-specifc-fields

hassandotcms committed:
Merge branch 'main' of https://github.com/dotCMS/core into feat/support-content-type-query-in-graphql

nollymar commented:
@hassandotcms , we use conventional commits to name PRs. For example: **_fix(core): Enable content-type-specific field queries in GraphQL page API_** or **_fix(graphql) Enable content-type-specific field queries in GraphQL page API_**. You can find more details [here](https://authoring.dotcms.com/devwiki/)

Also, the convention used for branch names is `issue-{issue_number}-more-details` or `{issue_number}-more-details`. In this case, the branch name could have been: `issue-32695-support-content-type-query-in-graphql` or `32695-support-content-type-query-in-graphql`

fabrizzio-dotCMS commented on dotcms-postman/src/main/resources/postman/GraphQLTests.json:
```diff
@@ -7832,6 +7832,300 @@
+								{
+									"key": "password",
+									"value": "admin",
+									"type": "string"
```
This should not be included here. I mean it is valid and it will work, but our postman are organized in such a way that they can use the json web token that is defined in auth on the parent element of the collection. 
If you initialize that element as a collection variable with a bearer token will be set. 

Authentication type must be Bearer token to stick with this pattern

fabrizzio-dotCMS commented on dotcms-postman/src/main/resources/postman/GraphQLTests.json:
```diff
@@ -7832,6 +7832,300 @@
+									"key": "password",
+									"value": "admin",
+									"type": "string"
+								}
```
Same here

fabrizzio-dotCMS commented on dotcms-postman/src/main/resources/postman/GraphQLTests.json:
```diff
@@ -7832,6 +7832,300 @@
+								{
+									"key": "password",
+									"value": "admin",
+									"type": "string"
```
And here

hassandotcms committed:
Merge branch 'main' of https://github.com/dotCMS/core into feat/support-content-type-query-in-graphql

hassandotcms committed:
test(graphql): update page field tests to use Bearer token auth

Replace hardcoded basic authentication with Bearer token pattern in new
GraphQL page field tests. Uses collection-level {{jwt}} token managed by
prerequest script for consistency with dotCMS Postman collection standards.

Refs: #32695

fabrizzio-dotCMS committed:
Merge branch 'main' into feat/support-content-type-query-in-graphql
